modules/apps/transfer/keeper/migrations.go

```diff
@@ -43,8 +43,9 @@ func (m Migrator) MigrateDenomMetadata(ctx sdk.Context) error {
 	m.keeper.iterateDenomTraces(ctx,
 		func(dt internaltypes.DenomTrace) (stop bool) {
 			// check if the metadata for the given denom trace does not already exist
-			if !m.keeper.bankKeeper.HasDenomMetaData(ctx, dt.IBCDenom()) {
-				m.keeper.setDenomMetadataWithDenomTrace(ctx, dt)
```

Comment:
If there exists a denom metadata we will not overwrite it. 

The only scenario in which this can overwrite an existing denom metadata, entering the if branch here is:
- `Collections/Map.Has` ([ref](https://github.com/cosmos/cosmos-sdk/blob/v0.50.6/collections/map.go#L98-L105)) returns `has == true` but also hits an error for some reason out of our control, meaning that `err != nil` in `HasDenomMetaData` [here](https://github.com/cosmos/cosmos-sdk/blob/v0.50.6/x/bank/keeper/keeper.go#L223-L226)

Reply:
I wonder if it is possible that the key constructed in an old version differs from the key constructed in a new version. We should verify that all non-ibc denom metadata's are queryable